### PR TITLE
Add ChangeAction to parse sub-options from --change

### LINKED ISSUE #1701: pypodman commit: TypeError: commit() missing 1 required positional argument: 'image_name'
**kind bug**

**Description**
`pypodman commit` does not work even without any options. 

**Steps to reproduce the issue:**
On remote machine:
```
root@localhost libpod]# podman version
Version:       0.10.1
Go Version:    go1.10.4
OS/Arch:       linux/amd64
```
```
[root@localhost libpod]# podman info
host:
  BuildahVersion: 1.5-dev
  Conmon:
    package: podman-0.10.1-1.gite4a1553.fc28.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.12.0-dev, commit: 452812fc5539855eb472129ff1132919a0e756be-dirty'
  Distribution:
    distribution: fedora
    version: "28"
  MemFree: 1119850496
  MemTotal: 2089267200
  OCIRuntime:
    package: runc-1.0.0-55.dev.git578fe65.fc28.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.0'
  SwapFree: 2147479552
  SwapTotal: 2147479552
  arch: amd64
cpus: 2
  hostname: localhost.localdomain
  kernel: 4.16.3-301.fc28.x86_64
  os: linux
  uptime: 4h 46m 36.86s (Approximately 0.17 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  ContainerStore:
    number: 2
  GraphDriverName: overlay
  GraphOptions:
  - overlay.mountopt=nodev
  - overlay.override_kernel_check=true
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
  Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 5
  RunRoot: /var/run/containers/storage
```
```
[root@localhost ~]# podman ps -a
CONTAINER ID   IMAGE                             COMMAND     CREATED          STATUS                      PORTS   NAMES              IS INFRA
dcbafa59eff0   docker.io/library/fedora:latest   /bin/bash   58 seconds ago   Exited (0) 40 seconds ago           heuristic_snyder   false
```
On host machine:
```
# hostnamectl
Static hostname: localhost.localdomain
Transient hostname: dhcp-2-7.nay.redhat.com
         Icon name: computer-laptop
           Chassis: laptop
        Machine ID: cd196ab74fbf40448d31037ceafbcf05
           Boot ID: 1e04aea17e044ac886ba0630c1b34dc1
  Operating System: Fedora 27 (Workstation Edition)
       CPE OS Name: cpe:/o:fedoraproject:fedora:27
            Kernel: Linux 4.18.7-100.fc27.x86_64
      Architecture: x86-64
```
```
python3 --version
Python 3.6.6
```
1. [install podman python3 package](https://github.com/containers/libpod/tree/master/contrib/python/podman)

2. [install pypodman](https://github.com/containers/libpod/tree/master/contrib/python/pypodman)

3. `pypodman -l root --host 192.168.122.28 commit heuristic_snyder image-committed-2`

**Describe the results you received:**
```
# pypodman -l root --host 192.168.122.28 commit heuristic_snyder image-committed-2

Traceback (most recent call last):
  File "/usr/local/bin/pypodman", line 11, in <module>
    load_entry_point('pypodman==0.0.0', 'console_scripts', 'pypodman')()
  File "/root/.local/lib/python3.6/site-packages/pypodman-0.0.0-py3.6.egg/pypodman/main.py", line 58, in main
  File "/root/.local/lib/python3.6/site-packages/pypodman-0.0.0-py3.6.egg/pypodman/lib/actions/commit_action.py", line 91, in commit
TypeError: commit() missing 1 required positional argument: 'image_name'
```

**Describe the results you expected:**
Can successfully create the image. 

**Additional information you deem important (e.g. issue happens only occasionally):**
` pypodman -l root --host 192.168.122.28 images` can work. On remote machine `podman commit` can properly create the image.

**Additional environment details (AWS, VirtualBox, physical, etc.):**
physical

### LINKED ISSUE #1702: pypodman commit: argument --change/-c: invalid choice
**kind bug**

**Description**
`pypodman commit` does not work with option `--change`.

**Steps to reproduce the issue:**
On remote machine:
```
root@localhost libpod]# podman version
Version:       0.10.1
Go Version:    go1.10.4
OS/Arch:       linux/amd64
```
```
[root@localhost libpod]# podman info
host:
  BuildahVersion: 1.5-dev
  Conmon:
    package: podman-0.10.1-1.gite4a1553.fc28.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.12.0-dev, commit: 452812fc5539855eb472129ff1132919a0e756be-dirty'
  Distribution:
    distribution: fedora
    version: "28"
  MemFree: 1119850496
  MemTotal: 2089267200
  OCIRuntime:
    package: runc-1.0.0-55.dev.git578fe65.fc28.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.0'
  SwapFree: 2147479552
  SwapTotal: 2147479552
  arch: amd64
cpus: 2
  hostname: localhost.localdomain
  kernel: 4.16.3-301.fc28.x86_64
  os: linux
  uptime: 4h 46m 36.86s (Approximately 0.17 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  ContainerStore:
    number: 2
  GraphDriverName: overlay
  GraphOptions:
  - overlay.mountopt=nodev
  - overlay.override_kernel_check=true
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
  Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 5
  RunRoot: /var/run/containers/storage
```
```
[root@localhost ~]# podman ps -a
CONTAINER ID   IMAGE                             COMMAND     CREATED          STATUS                      PORTS   NAMES              IS INFRA
dcbafa59eff0   docker.io/library/fedora:latest   /bin/bash   58 seconds ago   Exited (0) 40 seconds ago           heuristic_snyder   false
```
On host machine:
```
# hostnamectl
Static hostname: localhost.localdomain
Transient hostname: dhcp-2-7.nay.redhat.com
         Icon name: computer-laptop
           Chassis: laptop
        Machine ID: cd196ab74fbf40448d31037ceafbcf05
           Boot ID: 1e04aea17e044ac886ba0630c1b34dc1
  Operating System: Fedora 27 (Workstation Edition)
       CPE OS Name: cpe:/o:fedoraproject:fedora:27
            Kernel: Linux 4.18.7-100.fc27.x86_64
      Architecture: x86-64
```
```
python3 --version
Python 3.6.6
```
1. [install podman python3 package](https://github.com/containers/libpod/tree/master/contrib/python/podman)

2. [install pypodman](https://github.com/containers/libpod/tree/master/contrib/python/pypodman)

3. `pypodman -l root --host 192.168.122.28 commit --change CMD=/bin/bash --change ENTRYPOINT=/bin/sh --change LABEL=blue=image heuristic_snyder image-commited`

**Describe the results you received:**
```
# pypodman -l root --host 192.168.122.28 commit --change CMD=/bin/bash --change ENTRYPOINT=/bin/sh --change LABEL=blue=image heuristic_snyder image-commited
2018-10-22 19:56:02 CST | ERROR    | pypodman commit: argument --change/-c: invalid choice: 'CMD=/BIN/BASH' (choose from 'CMD', 'ENTRYPOINT', 'ENV', 'EXPOSE', 'LABEL', 'ONBUILD', 'STOPSIGNAL', 'USER', 'VOLUME', 'WORKDIR')
2018-10-22 19:56:02 CST | ERROR    | Try 'pypodman commit --help' for more information.
```

**Describe the results you expected:**
Can successfully create the image. 

**Additional information you deem important (e.g. issue happens only occasionally):**
` pypodman -l root --host 192.168.122.28 images` can work. On remote machine `podman commit` with the same options can properly create the image.

**Additional environment details (AWS, VirtualBox, physical, etc.):**
physical

### PR DESCRIPTION
**Pull request by jwhonce**
* Covers both commit and import commands
* Removed unneeded calls to super().__init__()

Fixes #1702
Fixes #1701

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: contrib/python/pypodman/pypodman/lib/parser_actions.py [hunk 1]
```diff
@@ -63,6 +64,53 @@ def __call__(self, parser, namespace, values, option_string=None):
...
+            metavar=metavar)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        """Convert and Validate input."""
+        items = getattr(namespace, self.dest, None) or {}
+        items = copy.copy(items)
+
+        choices = ('CMD', 'ENTRYPOINT', 'ENV', 'EXPOSE', 'LABEL', 'ONBUILD',
+                   'STOPSIGNAL', 'USER', 'VOLUME', 'WORKDIR')
+
```
Comment (TomSweeneyRedHat) [inline contrib/python/pypodman/pypodman/lib/parser_actions.py]: Is this supposed to emulate the functionality in `buildah config`?  If so, there are a number of other choices that we could potentially add. https://github.com/containers/buildah/blob/master/docs/buildah-config.md

**Comment by TomSweeneyRedHat** [comment]
Test failures, but me thinks they're flakes.  A question for consideration, otherwise LGTM with happy tests.

**Comment by baude** [comment]
LGTM

**Comment by baude** [comment]
@jwhonce please rebase and tests should pass

File: contrib/python/pypodman/pypodman/lib/parser_actions.py (hunk 1 above)
Comment (cevich) [reply contrib/python/pypodman/pypodman/lib/parser_actions.py]: @TomSweeneyRedHat he's just moving them from ``commit_action.py`` (above), nothing new here, only consolidated and made modular.

File: contrib/python/pypodman/pypodman/lib/parser_actions.py [hunk 2]
```diff
@@ -63,6 +64,53 @@ def __call__(self, parser, namespace, values, option_string=None):
...
+                 type=None,
+                 choices=None,
+                 required=False,
+                 help=None,
+                 metavar='OPT=VALUE'):
+        """Create ChangeAction object."""
+        help = (help or '') + ('Apply change(s) to the new image.'
+                               ' May be given multiple times.')
+
+        super().__init__(
```
Comment (cevich) [inline contrib/python/pypodman/pypodman/lib/parser_actions.py]: (Note: I don't know the code, nor remember if this is a python3 thing or not)

Doesn't this need to be ``super(ChangeAction, self).__init__(`` ?

Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/parser_actions.py]: @cevich Python3 finally made super calls simple as all classes are now "new" classes.

**Comment by jwhonce** [comment]
@baude I pulled the sub-options from the API.md.  Should I add all options from buildah.md?

File: contrib/python/pypodman/pypodman/lib/parser_actions.py (hunk 2 above)
Comment (cevich) [reply contrib/python/pypodman/pypodman/lib/parser_actions.py]: gotcha, thx.

**Comment by baude** [comment]
Probably so.. Would you like to stage it?

**Pull request merged**